cores/esp8266/core_esp8266_waveform_pwm.cpp

```diff
@@ -529,39 +532,39 @@ static IRAM_ATTR void timer1Interrupt() {
 
       // PWM state machine implementation
       if (pwmState.cnt) {
-        int32_t cyclesToGo;
+        int32_t pwmCyclesToGo;
         do {
-            cyclesToGo = pwmState.nextServiceCycle - GetCycleCountIRQ();
-            if (cyclesToGo < 0) {
-                if (pwmState.idx == pwmState.cnt) { // Start of pulses, possibly copy new
-                  if (pwmState.pwmUpdate) {
-                    // Do the memory copy from temp to global and clear mailbox
-                    pwmState = *(PWMState*)pwmState.pwmUpdate;
-                  }
-                  GPOS = pwmState.mask; // Set all active pins high
-                  if (pwmState.mask & (1<<16)) {
-                    GP16O = 1;
-                  }
-                  pwmState.idx = 0;
-                } else {
-                  do {
-                    // Drop the pin at this edge
-                    if (pwmState.mask & (1<<pwmState.pin[pwmState.idx])) {
-                      GPOC = 1<<pwmState.pin[pwmState.idx];
-                      if (pwmState.pin[pwmState.idx] == 16) {
-                        GP16O = 0;
-                      }
-                    }
-                    pwmState.idx++;
-                    // Any other pins at this same PWM value will have delta==0, drop them too.
-                  } while (pwmState.delta[pwmState.idx] == 0);
+          pwmCyclesToGo = pwmState.nextServiceCycle - GetCycleCountIRQ();
+          if (pwmCyclesToGo <= 0) {
+            if (pwmState.idx == pwmState.cnt) { // Start of pulses, possibly copy new
+              if (pwmState.pwmUpdate) {
+                // Do the memory copy from temp to global and clear mailbox
+                pwmState = *pwmState.pwmUpdate;
+              }
+              GPOS = pwmState.mask; // Set all active pins high
+              if (pwmState.mask & (1 << 16)) {
+                GP16O = 1;
+              }
+              pwmState.idx = 0;
+            }
+            else {
+              do {
+                // Drop the pin at this edge
+                GPOC = 1 << pwmState.pin[pwmState.idx];
```

Comment:
This bit was bracketed inside an `if` in the original code, so the logic here is different and I'm not sure correct.